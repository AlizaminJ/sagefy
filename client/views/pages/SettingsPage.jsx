import React from 'react'
import { Link } from 'react-router-dom'
import { shape, string } from 'prop-types'
import Layout from '../components/Layout'
import Icon from '../components/Icon'
import FormErrorsTop from '../components/FormErrorsTop'
import FormErrorsField from '../components/FormErrorsField'

export default function SettingsPage({ hash, gqlErrors, body: { id, name } }) {
  return (
    <Layout hash={hash} page="SettingsPage" title="Settings" description="-">
      <FormErrorsTop formErrors={gqlErrors} />
      <FormErrorsField formErrors={gqlErrors} field="all" />

      <section>
        <h1>
          Update your settings <Icon i="settings" s="xxl" />
        </h1>
        <p>
          <Link to="/password">
            <Icon i="password" /> Change my password
          </Link>
          .
          <br />
          or{' '}
          <Link to="/email">
            <Icon i="email" /> Change my email
          </Link>
          .
        </p>
        <form action="" method="POST">
          <input type="hidden" name="id" value={id} />
          <p>
            <label htmlFor="name">Name</label>
            <input
              id="name"
              name="name"
              placeholder="ex: Unicorn"
              type="text"
              size="40"
              value={name}
              required
            />
          </p>
          <FormErrorsField formErrors={gqlErrors} field="name" />
          {/* <p>
      <label for="email">Email</label>
      <input id="email" name="email" placeholder="ex: unicorn@example.com" type="email" size="40" value="doris@example.com">
    </p>
    <div>
      <label for="settings.email_frequency">Email Frequency</label>
      <ul class="ls-i" style="margin-top:0">
        <li><label><input type="radio" value="immediate" name="settings.email_frequency" checked> Immediate</label></li>
        <li><label><input type="radio" value="daily" name="settings.email_frequency"> Daily</label></li>
        <li><label><input type="radio" value="weekly" name="settings.email_frequency"> Weekly</label></li>
        <li><label><input type="radio" value="never" name="settings.email_frequency"> Never</label></li>
      </ul>
    </div> */}
          <p>
            <button type="submit">
              <Icon i="settings" /> Update settings
            </button>
          </p>
        </form>
      </section>
    </Layout>
  )
}

SettingsPage.propTypes = {
  hash: string.isRequired,
  gqlErrors: shape({}),
  body: shape({}),
}

SettingsPage.defaultProps = {
  gqlErrors: {},
  body: {},
}
