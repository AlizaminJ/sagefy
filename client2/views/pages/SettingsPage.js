/*

const merge = require('lodash.merge')
const { div, h1, p, a, hr } = require('../../helpers/tags')
const userSchema = require('../../schemas/user')
const { createFieldsData, findGlobalErrors } = require('../../helpers/forms')
const form = require('../components/form.tmpl')
const spinner = require('../components/spinner.tmpl')
const icon = require('../components/icon.tmpl')
const { getIsLoggedIn } = require('../../selectors/base')
const goLogin = require('../../helpers/go_login')

const fields = [
  {
    name: 'id',
    type: 'hidden',
  },
  {
    name: 'name',
    label: 'Name',
    placeholder: 'ex: Unicorn',
  },
  {
    name: 'email',
    label: 'Email',
    placeholder: 'ex: unicorn@example.com',
  },
  {
    name: 'settings.email_frequency',
    label: 'Email Frequency',
    options: [
      {
        label: 'Immediate',
      },
      {
        label: 'Daily',
      },
      {
        label: 'Weekly',
      },
      {
        label: 'Never',
      },
    ],
    inline: true,
  },
  {
    name: 'submit',
    type: 'submit',
    label: 'Update',
    icon: 'update',
  },
]

fields.forEach((field, index) => {
  fields[index] = merge({}, userSchema[field.name] || {}, field)
})

module.exports = data => {
  if (getIsLoggedIn(data) === null) {
    return spinner()
  }

  if (!getIsLoggedIn(data)) {
    return goLogin()
  }

  const user = data.users && data.users[data.currentUserID]
  if (!user) {
    return spinner()
  }

  const instanceFields = createFieldsData({
    schema: userSchema,
    fields,
    errors: data.errors,
    formData: merge(
      {},
      {
        id: user.id,
        name: user.name,
        email: user.email,
        'settings.email_frequency': user.settings.email_frequency,
      },
      data.formData
    ),
    sending: data.sending,
  })

  const globalErrors = findGlobalErrors({
    fields,
    errors: data.errors,
  })

  return div(
    { id: 'settings', className: 'page' },
    h1('Settings'),
    form({
      fields: instanceFields,
      errors: globalErrors,
    }),
    hr(),
    p(a({ href: '/password' }, icon('password'), ' Change my password.')),
    p(
      a(
        { href: 'https://gravatar.com' },
        icon('update'),
        ' Update my avatar on Gravatar.'
      )
    )
  )
}



const { getFormValues, parseFormValues } = require('../../helpers/forms')
const userSchema = require('../../schemas/user')

module.exports = (store, broker) => {
  const { getTasks } = store
  broker.add({
    'submit #settings form'(e, el) {
      if (e) {
        e.preventDefault()
      }
      let values = getFormValues(el)
      getTasks().updateFormData(values)
      const errors = getTasks().validateForm(values, userSchema, [
        'name',
        'email',
      ])
      if (errors && errors.length) {
        return
      }
      values = parseFormValues(values)
      getTasks().updateUser(values)
    },
  })
}

*/
