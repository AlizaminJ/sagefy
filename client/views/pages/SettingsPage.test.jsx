import SettingsPage from './SettingsPage'

describe('SettingsPage', () => {
  it('should render the SettingsPage', () => {
    expect(SettingsPage({ gqlErrors: {}, prevValues: {} })).toMatchSnapshot()
  })
})
