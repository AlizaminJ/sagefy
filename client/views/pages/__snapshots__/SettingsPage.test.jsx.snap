// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsPage should render the SettingsPage 1`] = `
<Layout hash={hash}
  description="-"
  page="SettingsPage"
  title="Settings"
>
  <FormErrorsTop
    formErrors={Object {}}
  />
  <FormErrorsField
    field="all"
    formErrors={Object {}}
  />
  <section>
    <h1>
      Update your settings
      <Icon
        i="settings"
        s="xxl"
      />
    </h1>
    <p>
      <Link
        replace={false}
        to="/password"
      >
        <Icon
          i="password"
          s="m"
        />
         Change my password
      </Link>
      .
      <br />
      or

      <Link
        replace={false}
        to="/email"
      >
        <Icon
          i="email"
          s="m"
        />
         Change my email
      </Link>
      .
    </p>
    <form
      action=""
      method="POST"
    >
      <input
        name="id"
        type="hidden"
      />
      <p>
        <label
          htmlFor="name"
        >
          Name
        </label>
        <input
          id="name"
          name="name"
          placeholder="ex: Unicorn"
          required={true}
          size="40"
          type="text"
        />
      </p>
      <FormErrorsField
        field="name"
        formErrors={Object {}}
      />
      <p>
        <button
          type="submit"
        >
          <Icon
            i="settings"
            s="m"
          />
           Update settings
        </button>
      </p>
    </form>
  </section>
</Layout>
`;
